sr_data_visualization/scripts/sr_data_visualization/sr_data_visualizer_gui.py

```diff
@@ -357,38 +332,37 @@ def make_all_button_functions(self, i, all, type):
             number_of_columns = 1
 
         if all:
-            def button_function(b):
+            def _button_function(b):
                 if b.text() == "All":
                     if b.isChecked():
-                        self.change_graphs(all=True, type=graph_type, ncol=number_of_columns)
+                        self._change_graphs(all=True, type=graph_type, ncol=number_of_columns)
         else:
             legend_name = self.global_yaml["graphs"][type]["lines"][i]
             legend_name_stripped = re.sub(r"[\(\[].*?[\)\]]", "", legend_name).strip()
 
-            def button_function(b):
+            def _button_function(b):
                 if legend_name_stripped in b.text():
                     if b.isChecked():
-                        self.change_graphs(all=False, legend_name=[legend_name],
-                                           line_number=i, type=graph_type, ncol=1)
-        return button_function
-
-    def change_graphs(self, all, **kwargs):
-        if kwargs["type"] == "pos_vel_eff":
-            index = 0
-        elif kwargs["type"] == "control_loops":
-            index = 1
-        elif kwargs["type"] == "motor_stat":
-            index = 2
+                        self._change_graphs(all=False, legend_name=[legend_name], 
+                                            line_number=i, type=graph_type, ncol=1)
+        return _button_function
+
+    def _change_graphs(self, all, **kwargs):
+        type_dict = {
```

Comment:
I think you can make the dictionary global variable in the init function defining it with self. and access it here and in previous function.

Reply:
I've done this, but it makes line 117 less clear. Let me know what you think